cpp/src/hash/hashing.cu

```diff
@@ -213,7 +221,168 @@ void compute_row_output_locations(size_type * __restrict__ row_partition_numbers
   }
 }
 
-template <bool has_nulls>
+/* --------------------------------------------------------------------------*/
+/** 
+ * @brief Move one column from the input table to the hashed table.
+ * 
+ * @param[in] input_buf Data buffer of the column in the input table
+ * @param[out] output_buf Preallocated data buffer of the column in the output table
+ * @param[in] num_rows The number of rows in each column
+ * @param[in] num_partitions The number of partitions to divide the rows into
+ * @param[in] row_partition_numbers Array that holds which partition each row belongs to
+ * @param[in] row_partition_offset Array that holds the offset of each row in its partition of
+ * the thread block.
+ * @param[in] block_partition_sizes Array that holds the size of each partition for each block
+ * @param[in] scanned_block_partition_sizes The scan of block_partition_sizes
+ */
+/* ----------------------------------------------------------------------------*/
+template <typename InputIter, typename DataType>
+__global__
+void move_to_output_buffer(InputIter input_iter,
+                           DataType *output_buf,
```

Comment:
Most likely irrelevant, but this one should be restricted too.

Reply:
Done